### PR TITLE
Proxypass setup

### PR DESCRIPTION
**Pull request by devzspy**
This will automatically setup the proxy pass protocols on port 80, 443 and 2222 from all edge nodes to middles and middles to proxies.

Resolves Issue #87

### CODE REVIEW

File: external/aws/output.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 output "ubuntu-20-04-latest-name" {
-  value = data.aws_ami.ubuntu.description
+  value = data.aws_ami.ubuntu
```
Comment (tophertimzen) [inline external/aws/output.tf:2]: This was changed to `value = data.aws_ami.ubuntu.description` in this commit https://github.com/redteaminfra/redteam-infra/commit/7663403bf6b2f054a02aa68fa10f126bf201e117, why are we reverting it?

File: external/aws/output.tf [hunk 2]
```diff
@@ -23,25 +22,18 @@ resource "local_file" "ansible_inventory" {
     proxies  = { for aws_instance in aws_instance.proxy: aws_instance.tags.Name => aws_instance.private_ip },
     elk      = { (aws_instance.elk.tags.Name) = aws_instance.elk.private_ip },
     username = var.image_username,
-    key      = local_sensitive_file.ssh_private_key.filename
+    key      = var.key_name
```
Comment (tophertimzen) [inline external/aws/output.tf:25]: This was changed in https://github.com/redteaminfra/redteam-infra/commit/6fcec82aa22d17858f3510fc87590c6ac8d59af4#diff-def107f03f40ee472e72195dd7c71414bd1b257e8e24a48ce97817ad7ec18f76, why is this commit reverting it?

File: external/aws/output.tf (hunk 1 above)
Comment (devzspy) [reply external/aws/output.tf:2]: Looks like whatever I had initially pulled from had what I am committing despite I never changed it. The only thing I had added was 

```
output "proxy_public_ips" {
  value = [for ip in aws_instance.proxy : ip.public_ip]
}
  ```

**Comment by devzspy** [comment]
Updated the output files to account for the commits that were inadvertently being reverted.

File: external/aws/output.tf [hunk 3]
```diff
@@ -23,25 +22,18 @@ resource "local_file" "ansible_inventory" {
...
   })
 }
 
-resource "local_sensitive_file" "ssh_private_key" {
-  content = tls_private_key.ssh_key.private_key_openssh
-  filename = pathexpand("~/.ssh/${var.engagement_name}")
-  file_permission = "0600"
```
Comment (willk) [inline external/aws/output.tf]: previous commit change being reverted

File: external/aws/output.tf [hunk 4]
```diff
@@ -23,25 +22,18 @@ resource "local_file" "ansible_inventory" {
...
+output "run-ansible" {
+  value = "Add your ssh users.yml file to ../../ansible/playbooks make any modification you need to site.yml, homebase.yml, proxies.yml, elk.yml, then run ansible\n\n\tcd ../../ansible && ansible-playbook -i inventory.ini site.yml\n"
 }
 
-resource "local_file" "ssh_public_key" {
-  content = tls_private_key.ssh_key.public_key_openssh
-  filename = pathexpand("~/.ssh/${var.engagement_name}.pub")
-  file_permission = "0600"
```
Comment (willk) [inline external/aws/output.tf]: previous commit change being reverted

File: external/sketch/ansible/roles/proxypass-setup/tasks/main.yml
Comment (willk) [inline external/sketch/ansible/roles/proxypass-setup/tasks/main.yml:1]: What happens when you want to change ports, or have multiple edges go to one middle and to one proxy?

**Comment by willk** [comment]
Let's use ansible for all the configuration over running a python script that gets added to the host. 

Make this specific to tcp connection back in (just ssh in this case).

Your server block should look similar to: [stream-template.conf.j2](https://github.com/redteaminfra/redteam-infra/blob/nginx-untrusted-redirector/ansible/roles/nginx-untrusted-redirector/templates/stream-template.conf.j2). Note that `proxy_protocol on;` is only needed if you need to pass proxy information on to a host down the line. I could be useful for logging, but I'm not sure if it is compatible with ssh.

You should also ensure that nginx is installed on edges and middles. Currently that is part of configuring edges and middle; however, that is not guaranteed in the future.

**Comment by devzspy** [comment]
Updates made in latest commit

**Comment by willk** [comment]
Slow to review, but I haven't forgotten about it

**Review by willk — APPROVED** [review]
Didn't test end-to-end, but didn't break AWS standing it up. lgtm

**Pull request merged**